### PR TITLE
RT: add base departure/arrival in journey

### PR DESCRIPTION
**Pull request by pbougue**
JIRA: http://jira.canaltp.fr/browse/NAVP-342

### CODE REVIEW

**Comment by pbougue** [comment]
Check submodule before merge

File: source/routing/raptor_api.cpp [hunk 1]
```diff
@@ -253,6 +253,37 @@ static void add_direct_path(PbCreator& pb_creator,
...
+        auto logger = log4cplus::Logger::getInstance("log");
+        LOG4CPLUS_DEBUG(logger, "Ignored stop_time finding: impossible to match exactly one base stop_time");
+        return nullptr;
+    }
+    return st_base;
+}
+
+static bt::ptime get_base_dt(const nt::StopTime* st_orig, const nt::StopTime* st_base,
+                             const bt::ptime& dt_orig, bool is_departure) {
+    auto validity_pattern_dt_day = dt_orig.date();
```
Comment (antoine-de) [inline source/routing/raptor_api.cpp:288]: a bit of comments here would be nice

File: source/routing/raptor_api.cpp [hunk 2]
```diff
@@ -253,6 +253,37 @@ static void add_direct_path(PbCreator& pb_creator,
...
+    validity_pattern_dt_day -= boost::gregorian::days(st_orig->vehicle_journey->shift);
+    auto hour_of_day_orig = (is_departure ? st_orig->departure_time : st_orig->arrival_time);
+    validity_pattern_dt_day -= boost::gregorian::days(hour_of_day_orig / (60*60*24));
```
Comment (antoine-de) [inline source/routing/raptor_api.cpp]: it's better to use SECONDS_PER_DAY

**Comment by TeXitoi** [comment]
OK for me with tests.

**Pull request merged**